### PR TITLE
Arcade rewards red space suit helmet replica again

### PR DESCRIPTION
**Pull request by ghost**
Arcade rewards red space suit helmet replica again.

### CODE REVIEW

**Comment by jakmak6** [comment]
Why was it removed in the first place?

**Comment by Kurfursten** [comment]
I looked through the commit history for the file and can find no record that you could ever get the helmet from the arcade, as far back as three years ago

Seems harmless though, maybe you should get both in a bundle instead of just the helmet or suit though.

**Comment by ghost** [comment]
@Kurfursten The whole thing near as I can tell is is that it's supposed to come with the suit, but because of how it's ordered, it spawns the suit and exits before it reaches the part that says hey, you rolled the suit so spawn the helmet too. Or that it simply wasn't carried over to the second droplist.  It's a bandaid but it works. Should I go ahead and just make a box for em and toss that in instead?

`
/obj/machinery/computer/arcade/proc/arcade_action()
	if ((src.enemy_mp <= 0) || (src.enemy_hp <= 0))
		if(!gameover)
			src.gameover = 1
			src.temp = "[src.enemy_name] has fallen! Rejoice!"

			if(emagged)
				feedback_inc("arcade_win_emagged")
				new /obj/item/clothing/head/collectable/petehat(src.loc)
				new /obj/item/device/maracas/cubanpete(src.loc)
				new /obj/item/device/maracas/cubanpete(src.loc)
				message_admins("[key_name_admin(usr)] has outbombed Cuban Pete and been awarded explosive maracas.")
				log_game("[key_name_admin(usr)] has outbombed Cuban Pete and been awarded explosive maracas.")
				src.New()
				emagged = 0

			else if(!contents.len)
				feedback_inc("arcade_win_normal")
				var/prizeselect = pickweight(prizes)
				new prizeselect(src.loc)

				if(istype(prizeselect, /obj/item/toy/gun)) //Ammo comes with the gun
					new /obj/item/toy/ammo/gun(src.loc)

				else if(istype(prizeselect, /obj/item/clothing/suit/syndicatefake)) //Helmet is part of the suit
					new	/obj/item/clothing/head/syndicatefake(src.loc)

			else //admins can varedit arcades to have special prizes via contents, but it removes the prize rather than spawn a new one
				feedback_inc("arcade_win_normal")
				var/atom/movable/prize = pick(contents)
				prize.forceMove(src.loc)`

**Comment by Kurfursten** [comment]
Yeah, a box would be the cleanest solution without these weird exceptions.

**Comment by ghost** [comment]
Alright, later tonight I'll box em and test it.

**Comment by ghost** [comment]
Alrighty, works. Crossbow and crayon drop too so I don't think the long address messed with anything.

untitled

Edit: Have a question regarding the return. Some of the boxes are using return at the end and others aren't when populating the box. Which is the proper version to use?

**Comment by Dacendeth** [comment]
Thoughts on using a syndicate red box for the suit, for thematic reasons?

File: code/game/objects/items/weapons/storage/boxes.dm [hunk 1]
```diff
@@ -553,6 +553,15 @@
...
 		new /obj/item/toy/snappop(src)
 
+/obj/item/weapon/storage/box/syndicatefake/space
+	name = "Space Suit and Helmet Replica"
+
+/obj/item/weapon/storage/box/syndicatefake/space/New()
+	..()
+	new /obj/item/clothing/suit/syndicatefake(src)
+	new /obj/item/clothing/head/syndicatefake(src)
+	return
```
Comment (Kurfursten) [inline code/game/objects/items/weapons/storage/boxes.dm]: Don't let PJB find out about this trailing return

You were right to ask about it of course, from a performance perspective it has no effect but *some people* are quite bothered by them.

Comment (ghost) [reply code/game/objects/items/weapons/storage/boxes.dm]: Alright, so it basically doesn't do anything and should get the boot?

Comment (Intigracy) [reply code/game/objects/items/weapons/storage/boxes.dm]: Ye

**Comment by ghost** [comment]
There we are, trailing return's gone and I actually found the syndiebox sprite this time.
untitled

**Pull request merged**